lazy-pages/src/sys/unix.rs

```diff
@@ -29,24 +30,41 @@ thread_local! {
     static OLD_SIG_HANDLER: RefCell<SigHandler> = RefCell::new(SigHandler::SigDfl);
 }
 
-extern "C" fn handle_sigsegv(sig: i32, info: *mut siginfo_t, ucontext: *mut c_void) {
-    unsafe {
-        #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
-        let is_write = {
-            let ucontext = ucontext as *const nix::libc::ucontext_t;
+cfg_if! {
+    if #[cfg(all(target_os = "linux", target_arch = "x86_64"))] {
+        unsafe fn ucontext_get_write(ucontext: *mut nix::libc::ucontext_t) -> Option<bool> {
             let error_reg = nix::libc::REG_ERR as usize;
             let error_code = (*ucontext).uc_mcontext.gregs[error_reg];
             // Use second bit from err reg. See https://git.io/JEQn3
             Some(error_code & 0b10 == 0b10)
-        };
+        }
+    } else if #[cfg(all(target_os = "macos", target_arch = "aarch64"))] {
```

Comment:
and what if we use macos x86?

Reply:
`None` is returned then. Also I don't have x86 macOS machine to implement the logic for it

Reply:
> `None` is returned then. Also I don't have x86 macOS machine to implement the logic for it

Do you mean that this is still opened issue, right?

Reply:
Yes